src/Http/Http/src/Internal/MutableFormOptionsMetadata.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Microsoft.AspNetCore.Http.Metadata;
+
+namespace Microsoft.AspNetCore.Http.Features;
+
+internal class MutableFormOptionsMetadata(IFormOptionsMetadata formOptionsMetadata) : IFormOptionsMetadata
```

Comment:
Actually, why is this even a class, looks like we could just new up a FormOptions and set the values on it when merging.

Reply:
We can't use `FormOptions` directly because we want to preserve the `null` values for each property as we do the merge. For example, for the following piece of code:

```
var group = app.MapGroup("/group")
  .WithFormOptions(valueCountLimit: 2);
group.MapPost("/").WithFormOptions(valueCountLimit: 7, valueLengthLimit: 3);
```

We'll get the following metadata populated in our endpoint:

```
{ valueCountLimit: 2, valueLengthLimit: null }
{ valueCountLimit: 7, valueLengthLimit: 3 }
```

When we merge, we favor the property on the current metadata over the previou _unless_ it's null. If we started with `FormOptions` all the values are required and we'd lose the ability to apply the merge correctly.

Reply:
Could be a struct still, to avoid the class alloc since we just throw it away afterwards.

Reply:
Yep! Good call.